asdf_astropy/converters/transform/mappings.py

```diff
@@ -0,0 +1,111 @@
+from .core import TransformConverterBase
+
+
+from asdf.extension import Converter
+
+
+class IdentityConverter(TransformConverterBase):
+    """
+    ASDF support for serializing the Identity model.
+    """
+    tags = ["tag:stsci.edu:asdf/transform/identity-*"]
+
+    types = ["astropy.modeling.mappings.Identity"]
+
+    def to_yaml_tree_transform(self, model, tag, ctx):
+        node = {}
+        if model.n_inputs != 1:
+            node['n_dims'] = model.n_inputs
+        return node
+
+    def from_yaml_tree_transform(self, node, tag, ctx):
+        from astropy.modeling.mappings import Identity
+
+        return Identity(node.get('n_dims', 1))
+
+
+class RemapAxesConverter(TransformConverterBase):
+    """
+    ASDF support for serializing the Mapping model
+    """
+    tags = ["tag:stsci.edu:asdf/transform/remap_axes-*"]
+
+    types = ["astropy.modeling.mappings.Mapping"]
+
+    def to_yaml_tree_transform(self, model, tag, ctx):
+        node = {"mapping": list(model.mapping)}
+        if model.n_inputs > max(model.mapping) + 1:
+            node["n_inputs"] = model.n_inputs
+        return node
+
+    def from_yaml_tree_transform(self, node, tag, ctx):
+        from astropy.modeling.models import Mapping
+
+        return Mapping(tuple(node["mapping"]), node.get("n_inputs"))
+
+
+class UnitsMappingConverter(Converter):
+    """
+    ASDF support for serializing the UnitsMapping model.
+    Note that this converter does not inherit from TransformConverterBase,
+    because the inputs and outputs are written differently
+    from other models.
```

Comment:
Is the reason the inputs and outputs are written different because of the way the schema is structured?  If so, would it be possible to modify the schema so this becomes more standard?  Or is it just a fundamentally different serialization?

Reply:
This model is funny, it's not even really a model but was convenient to implement that way.  It's different enough from the other models that most of the common properties don't make sense -- it would never have a name, an inverse, constraints, bounds, etc.